docs/dev/developer-guides/contracts/system-contracts.md

```diff
@@ -35,3 +35,8 @@ For greater extensibility and to lower the overhead, some parts of the protocol
 
 For now, you do not have to know any details about it, but knowing that it exists is important when you develop using the account abstraction feature. You can always assume that the bootloader is not malicious and it is a part of the protocol. In the future, the code of the bootloader will be made public and any changes to it will also mean an upgrade to the protocol.
 
+## Protected access to some of the system contracts
+
+Some of the system contracts have impact on the account that may not be expected on Ethereum. For instance, on Ethereum the only way EOA account could increase its nonce is by sending a transaction. Also, sending a transaction could only increase nonce by 1 at a time. On zkSync nonces are implemented via the [NonceHolder](#inonceholder) system contract and, if naively implemented, the users could be allowed to increment their nonces by calling this contract. That's why the calls to most of the non-view methods of the nonce holder were restricted to be called only with a special `isSystem` flag, so that interactions with important system contracts could be conciously managed by the developer of the account.
```

Comment:
It's quite confusing. Is `isSystem` an indicator of extended ABI required by e.g. ETH transfer simulator, or is it just a check over `msg.sender <= u16::MAX`?